### ISSUE TITLE
Move's Attack Power

### ISSUE THREAD

**Issue by an8bitbatty** [SunakazeKun/PMDe#5]
Hello, I've been using your editor for some time now, & I absolutely love it, I've been editing PMD for a project on Amino, & I needed to know how I can edit the Attack Power of Moves because I changed all Pokémon's stats to match that of the Main-Series games, & now all the attacks do only 1 point of damage (except for critical hits, which do 2) I would like to know, because as great as PMDe is, it's not as user-friendly as I had hoped, but that's the only issue I'm finding with it, How exactly can I edit the Power-Level of Moves? (The game itself doesn't even tell the player how strong the move is either) (And in the Editor, all the Moves that do damage other than the Attack command say 99 under the Power section, so is that the power-level of the move? or is it something else?, (Which Factor determines the strength of the move?))

**Comment by shadowlucario50** [comment]
> I've been editing PMD for a project on Amino, & I needed to know how I can edit the Attack Power of Moves because I changed all Pokémon's stats to match that of the Main-Series games, & now all the attacks do only 1 point of damage ... How exactly can I edit the Power-Level of Moves? (In the Editor, all the Moves that do damage other than the Attack command say 99 under the Power section, so is that the power-level of the move? or is it something else?)

That's just the mechanics of the game. All attacking moves are at the same attack power because it's developed that way. If you wanted to give it the main-game power, you would have to rework the entire programming of the system since, as far as I know, Move Power is a percentage of the Pokémon's potential, or the Pokémon's level and stats. The level itself effects some hidden stats to make your Pokémon stronger or weaker, and the stats themselves determine how powerful the attack would be. There's also the RNG rolls of if it will be a strong or weak attack each turn.

I wouldn't know how much damage would be reduced when lowering the Power though. Maybe I'm wrong and it's actually drops the max damage down from 100 (ex. 85 power would take away 15 from the max damage). I'm just assuming how it works based on observation, especially since attack power can be registered between -128 and 127. I don't know what to make of that except it's a signed char (almost, normal is -127 to 127), and I don't understand much about that either...

**Edit:** Upon further investigation, there's more than meets the eye. It apparently doesn't do much of anything on lower levels. Here are some of the damage results I gathered below when changing the power of a move...

Poochyena Lv100 with 113 Attack using Tackle in Tiny Woods
99 - 522, 547, 534, 518, 549
0 - 494, 522, 496, 453, 448
-99 - 524, 536, 477, 502, 545

Charmander Lv9 with 17 Attack using Scratch in Tiny Woods
99 - 25, 22, 21, 22, 22
40 - 27, 30, 26, 24, 22
0 - 27, 23, 20, 26, 26
-99 - 25, 22, 24, 27, 23

On Poochyena, it has a more significant change, becoming lower when it nears 0. However, the opposite was true with Charmander... Either results are faulty, or there's a more complicated equation to Power than I thought.

However, that brings me to question what happened to your hack to make it only do 1 damage per hit... Did you edit the stats of the other Pokémon as well?

**Comment by an8bitbatty** [comment]
Yes, I edited every Pokémon's Base Stats to match that of the main game, so where most of their stats would normally be a 1 digit number, most of them are now 2-3 digit numbers, for instance, If you take a look at Chikorita's stats at level 5 & Sunkern's stats at whatever level it appears as in Tiny Woods, It's much higher than they normally are, (anywhere from 20-70 in each stat) But after reading what you said ^up there^, The moves shouldn't have changed in initial attack power, & should've been calculated by the Pokémon's stats instead, but they're not, I tested it out in a completely new save file, so that any data from a previous save where the stats would remain the same wouldn't result in corruption or anything like that. The only thing I can think of is that it's trying to take the attacking Pokémon's Attack Stat, & the defending Pokémon's Defense stat, & it's not weighing the difference when calculating how strong the move would be. I didn't understand a lot of that, & even my own words are beginning to confuse me, but I'll give you a copy of the .gba file & if you can take a look at it & see what I may have done wrong, (or if it works for you) I could try moving some files & see if any ghost .savs are corrupting the game* I understand that that the Attacks all range from 127 to -128, but I don't understand what that has to do with them all being pre-set to 99, or for that matter why all of the Pokémon's base stats are so low & incorrect to what they should be when the limit for them all is 65535* (So, should I increase all the attacking moves from 99 to a higher number in order for them to hit for full power?)

Edit: I'm not sure if this matters, but I edited all the base stats in version 1.02(b) of the editor before the interface was updated, but with my technician's hat on I'd say that has nothing to do with the stats & attacks*

(Unrelated, but I'd still like to make it work): I tried to implement Munchlax into the game, but when I was testing it in a dungeon, it wasn't gaining stats upon level up (Except for HP), & the stats were the same as the base stats at level 5. It also only had 1 move "Item Toss" & It would be an inconvenience if it can't learn any TMs/HMs to compensate for this. I will try to edit the base stats & see if they do anything different (actually gain stats upon level up).

Edit: I have discovered that not even changing the base stats allows for Munchlax's stats to increase upon level up, so it might have something to do with another setting that I don't know how to modify, this is saddening, I wanted to play as "Chowder", Oh well. I have to remove it from Howling woods now, & remove all the information I added for it.

Thanks for getting back to me so soon, you have no idea how much this means to me, Thankyou!

**Comment by SunakazeKun** [comment]
Never ever provide ROM files! It's illegal. Also, I'll look into this more asap.

**Comment by an8bitbatty** [comment]
Sorry, but I thought if I provided the results of my tinkering with the rom, it'd be easier to solve the problem, since I had already done all the work to edit every single Pokémon's stats* (Incase anybody else had this idea in the future) It's ok if you can't fix it, I have come to terms with the possibility that it won't work, & am comfortable with a normal version* But I feel like this is a serious issue for people who want to edit the stats on such a broad scale as this, It was actually quite confusing to myself, as I have a basic understanding of code. It is curious why the damage formula is wrong. Thanks for getting back to me on such short notice, & again I'm sorry, I just thought it'd help resolve the issue.

Edit: I just opened up the PMD: RRT Rom in the Editor & to my surprise, many of the Pokémon had their walkable tiles altered to something completely wrong, like Water types all now walk on magma, & fire on water, etc. I'm now having to change them all back, I think this has something to do with the interface being updated, & extra entries being added*

Edit: Correction, The Pokémon can still walk on the appropriate tiles. However, They are displayed incorrectly in the editor as a different entry* (IE: Magma is Water & Vice-Versa) Glad I had a zip file to back up the edits I made to the Walkable Tiles, It was something else entirely! ( '^-^) (I think adding the "Wall" option to the Walkable Tiles drop-down menu caused the other options to shuffle down & essentially swap with another option In-game*)

**Comment by SunakazeKun** [comment]
Yup, that walkable tiles thing will be fixed in the next build. Thanks for pointing out.

**Comment by an8bitbatty** [comment]
I just noticed something, & before the 1.03 Update it was working, I made it to where I could bring 4 Pokémon into a dungeon instead of 3, but now It tells me I have too many* I'll try to find the 1.02 version & try to fix it in the previous version* (It now tells me that there are too many Pokémon in the party & must be reduced by 1, but it didn't do that before)

Edit: Trying to fix it in either version didn't work, I think the game has somehow locked the .sav file out of trying to bring 4 Pokémon into a dungeon, by attempting to go on an escort mission, I think* Because I didn't seem to have any problem before. Before I started a new game I was able to bring 4 members into a dungeon*

Edit: Also, the Party Limit capped at 4 Members in version 1.02, but now the Party Limit caps at 40, could that have something to do with it? because when I was editing the .gba file in version 1.02, it worked* I think it may have been altered in a way that undid the ability to bring more than 3 Pokémon into a dungeon* & I know for a fact that it worked in version 1.02 because I was able to bring in all of Skitty, Cyndaquil, Mawile, & Porygon into a dungeon at once, but I think trying to edit it in version 1.03 of PMDe broke it somehow?

Edit: And one more thing, I realized we can change the amount of items we can BRING into a dungeon, but what about the Treasure Bag in the Town? Is there a way to edit how may Items we can have in the bag at any given time? Because I didn't think it was possible to change that number*

**Comment by shadowlucario50** [comment]
> I just noticed something, & before the 1.03 Update it was working, I made it to where I could bring 4 Pokémon into a dungeon instead of 3, but now It tells me I have too many* I'll try to find the 1.02 version & try to fix it in the previous version* (It now tells me that there are too many Pokémon in the party & must be reduced by 1, but it didn't do that before)
>
> Edit: Trying to fix it in either version didn't work, I think the game has somehow locked the .sav file out of trying to bring 4 Pokémon into a dungeon, by attempting to go on an escort mission, I think* Because I didn't seem to have any problem before. Before I started a new game I was able to bring 4 members into a dungeon*
>
> Edit: Also, the Party Limit capped at 4 Members in version 1.02, but now the Party Limit caps at 40, could that have something to do with it? because when I was editing the .gba file in version 1.02, it worked* I think it may have been altered in a way that undid the ability to bring more than 3 Pokémon into a dungeon* & I know for a fact that it worked in version 1.02 because I was able to bring in all of Skitty, Cyndaquil, Mawile, & Porygon into a dungeon at once, but I think trying to edit it in version 1.03 of PMDe broke it somehow?

Honestly, I think you should make a new thread for these things and not use the same one since these aren't relative to the first topic of this post. It might make it more convenient for SunakazeKun...

Anyway, the 4 Pokémon limit only appears for continuation dungeons like Mt. Blaze to Mt. Blaze Peak, Frosty Forest to Frosty Grotto, Sky Tower to Sky Tower Summit, etc. Putting it at 4 might make the missions break, since no dungeon with the 4 Pokémon limit have any missions. Then again, Meteor Cave doesn't have any missions, so I cannot say.

Also, the cap's at 40? What...?

**Edit:** The cap from 5-40 does nothing as the max party members are set to 4. You cannot even recruit more inside the dungeon. Also, I tested it and it still works on both a clean file and my own hacked file. You may have changed something in a dungeon that broke the code for your max. I'm using v1.03, and I can bring more Pokémon into the dungeon. I tested for escort missions, and it has a message stopping you if have too many Pokémon in your party.

> Edit: And one more thing, I realized we can change the amount of items we can BRING into a dungeon, but what about the Treasure Bag in the Town? Is there a way to edit how may Items we can have in the bag at any given time? Because I didn't think it was possible to change that number*

That's a different line of code, but I'm certain there could be an increase and decrease to the number of items for the bag. However, the game would need some way to handle the pages of these items, so viewing it would be more difficult.

**Comment by an8bitbatty** [comment]
See, I used to be able to bring 4 Pokémon into any & all Dungeons (Before the 1.03 update) Including the Dojo(s), By changing the Party Limit from 3 to 4, but now I get messages like this* I don't see why it doesn't work now, when it did before, (I didn't even get "error" messages when I tried to go on escort missions before), but what I think happened is by changing the Party Limit cap from 4 (1.02) to 40 (1.03), It no longer allows this **IMPORTANT** Quality of life Improvement from working* I would very much like to be able to do this once again, as it increased the value of the entire experience greatly! Especially once I got 4 Permanent Party Members that I didn't plan on switching out any time soon* Thankyou!
![screenshot_2017-10-11-20-30-18](https://user-images.githubusercontent.com/32327332/31478502-c4b53462-aede-11e7-99ac-ed51145f80c5.png)

**Comment by shadowlucario50** [comment]
Maybe more pictures, because that's the default message for the dojo if you try to bring 4 members there, and editing the dojo is still a little finicky. Could you post a picture for when you attempt a dungeon?

**Comment by an8bitbatty** [comment]
@shadowlucario50 
Sure, as you can see here:
![screenshot_2017-10-12-12-29-38](https://user-images.githubusercontent.com/32327332/31507902-5dafd2b0-af4a-11e7-98a3-e933a216a081.png)
I have a full party of 4 Pokémon. But when I try to go into a dungeon:
![screenshot_2017-10-12-12-30-01](https://user-images.githubusercontent.com/32327332/31507932-7b971874-af4a-11e7-80e8-391fc27472f5.png)
I get this message, when before, it would allow me to enter, regardless of having 4 Pokémon in the Party*
And as you can see in this image, I am using Version 1.03 of PMDe:
![1012171232](https://user-images.githubusercontent.com/32327332/31508092-01444e1a-af4b-11e7-8b49-6da0dd821094.jpg)
And when I'm editing the Party Limit of Dungeons in THIS version, as apposed to version 1.02, the Party Limit caps out at 40, instead of 4: (the video had to be compressed in order for me to upload it)
[1012171233.zip](https://github.com/SunakazeKun/PMDe/files/1380034/1012171233.zip)
I believe that, (similar to the mistake made when SunakazeKun updated the "Walkable Tiles" section in the Pokémon Data Editor), This had something to do with it, as before I updated my copy of the Editor, This Change still worked* (Being able to bypass the Default Party Limit) I'm sure you'll find that if you tried to edit a vanilla copy of PMD: Red Rescue Team in Version 1.02 & set all the Dungeon's Party Limits to 4, it **would** work*
**Edit:** How would I have broken the code for the max party limit? I didn't know that the Editor could break the code of the game, just alter it at will & revert to what it was before* I haven't edited anything in the dungeon section since version 1.02, & I also got a message stopping me from bringing 4 Pokémon on an escort mission, (when I accidently forgot I had an escort mission in the list), but the first time, it was also followed by another message*, So I had 4, tried to go on an escort Mission, & that's when it stopped me with 2 messages, did you attempt to go in with 4 Pokémon already in your party, or less? In the Editor in Both Versions, the Party Limit is set to 4, & has been since Version 1.02, could updating with the Party Limit **already** being set to 4 have something to do with it? Because I changed all 3-limit dungeons to 4 in the previous version, & it remained that way when I updated. **I even started a new game!** With the way you put it, it sounds like I'm looking for a needle in a haystack at this point.
**Edit:** If it's only something to do with the Dungeons, (& if it's probably just some of them don't read well, having a 4 Pokémon Limit), Do you think you could take a look at it in PMDe for me? You'd probably know a lot more about it than I would... ( '._.)
https://www.dropbox.com/s/cn8r4zpf3xj1n0o/Pokemon%20Mystery%20Dungeon%20-%20October%20Edit.zip?dl=0
I'll also provide the .sav file so you can see, along side the edits I have made, that it won't let me bring 4 Pokémon into any dungeon, even though all of them (that normally allow 3) are set to 4*
https://www.dropbox.com/s/3a3gxhikg9bmf0k/Pokemon%20Mystery%20Dungeon%20-%20October%20Edit.sav?dl=0
The only other thing I can think of that could be the problem is something went wrong with the .sav file, & either broke or corrupted the Party Limit change*

**Comment by shadowlucario50** [comment]
You know, you really should avoid messing with the Corrupted dungeons... :< Causes lots of problems. I don't know how to fix this. I tried to change all back to 3 to see if I could make it back to 4, but that didn't fix it. When editing the Corrupted dungeons, who knows what can happen?

**Comment by an8bitbatty** [comment]
@shadowlucario50 
You just gave me an idea, I know that I had made all of the Dungeon's Party Limits 4 in Version 1.02, & all of them still worked, but what if that's the problem **NOW** in version 1.03? I think I should try this: Take the .gba file as it is now, open it up in Version 1.02, fix all dungeons to 3, Start a new game, change all the Non-Corrupted dungeons to 4 in Version 1.03, & see if it works then! (It's not just me that has to worry about this bug), I made this Rom Edit for me & a bunch of other people, built on community feedback & requests, & this is one of the first Quality of Life improvements I promised to them* So I'll do whatever it takes to resolve this issue! Thankyou for getting back to me so soon, If you find any way of resolving this issue please let me know, you've been a great help so-far & I wish I had even half the brains you've shown to have! ( '^w^) I think the big issue right now is the fact that I may have a corrupted .sav file all-together & even if I fixed the issue somehow, the file could still not work properly* So I'll try everything, I didn't know how catastrophic messing around with the corrupted dungeons could be to the entire game! ( '>w<) I'll remember to listen to obvious warnings like that in the future! Again, Thanks for all the help you've been giving me, if you can help me resolve this issue somehow I couldn't thank you enough! (>^w^<)

**Comment by SunakazeKun** [comment]
Actually, the party limit shouldn't go beyond 4. Because of a stupid copy-paste mistake I made during reworking the UI the limit could be set to 40...

**Comment by an8bitbatty** [comment]
I have one more question regarding this issue, when editing the Moves, and say for instance I'm editing the move Flash, & I wanted it to do damage as well as lower the opponent's accuracy, how would I go about applying a Damage to this Move? (I'd like to make it a Psychic-Type Attacking Move that Lowers all Foe's Accuracy within a 1-tile range)
